backend/src/oe/api/op-blockspan.service.ts

```diff
@@ -0,0 +1,141 @@
+      throw new Error(`${UUID} OpBlockSpanApiService.$getBlockSpanList: error while generating block span list: ${e instanceof Error ? e.message : e}`);
+    }
+  }
+
+  private async $getConfirmedBlockSpanList(UUID: string, blockSpanList: BlockSpan[]): Promise<ConfirmedBlockHeight[]> {
+    try {
+      const confirmedBlocks = [] as ConfirmedBlockHeight[];
+      const currentTip = await bitcoinApi.$getBlockHeightTip();
+
+      blockSpanList
+        .filter((blockSpan: BlockSpan) => (
+          opBlockHeaderService.isBlockHeightVerified(blockSpan.startBlockHeight, currentTip)
+          && opBlockHeaderService.isBlockHeightVerified(blockSpan.endBlockHeight, currentTip)
+        ))
+        .forEach((blockSpan: BlockSpan) => {
+          confirmedBlocks.push(
+            { value: blockSpan.startBlockHeight },
+            { value: blockSpan.endBlockHeight },
+          );
+        });
+
+      return confirmedBlocks;
+    } catch (e) {
+      throw new Error(`${UUID} OpBlockSpanApiService.$getConfirmedBlockSpanList: error while filtering confirmed block span list: ${e instanceof Error ? e.message : e}`);
+    }
+  }
+
+  public async $generateBlockSpanDetailedList(UUID: string, endBlockHeight: number, span: number, blockSpanCount: number): Promise<BlockSpanDetails[]> {
+    try {
+      const blockSpanList = this.$getBlockSpanList(
+        UUID,
+        endBlockHeight,
+        span,
+        blockSpanCount
+      );
+      const confirmedBlockSpanList = await this.$getConfirmedBlockSpanList(UUID, blockSpanList);
+      if (!confirmedBlockSpanList.length) {
+        return [];
+      }
+
+      const blockHeadersList = await opBlockHeaderService.$getBlockHeadersByHeights(
+        UUID,
+        Array.from(new Set(confirmedBlockSpanList))
+      );
+
+      let index = 0;
+      const result: BlockSpanDetails[] = [];
+      while (index < blockHeadersList.length - 1) {
+        result.push({
+          startBlock: {
+            height: blockHeadersList[index].height,
+            hash: blockHeadersList[index].current_block_hash,
+            mediantime: blockHeadersList[index].mediantime,
+            timestamp: blockHeadersList[index].timestamp,
+          },
+          endBlock: {
+            height: blockHeadersList[ ++index ].height,
+            hash: blockHeadersList[index].current_block_hash,
+            mediantime: blockHeadersList[index].mediantime,
+            timestamp: blockHeadersList[index].timestamp,
+          },
+        });
+      }
+      return result;
+    } catch (e) {
+      throw new Error(`${UUID} OpBlockSpanApiService.$generateBlockSpanDetailedList: error while generating detailed block span list: ${e instanceof Error ? e.message : e}`);
+    }
+  }
+
+  public async $getBlockSpanDetailedList(
+    UUID: string,
+    endBlockHeight: number,
+    span: number,
+    blockSpanCount: number,
+    withNbdrStatistics: boolean
+  ): Promise<BlockSpanDetails[] | BlockSpanDetailsWithNbdr[] | NbdrStatisticsError> {
```

Comment:
Today I tried to evaluate this function for a second time and came to conclusion, that I have hard time of understanding what am I expecting as the result: maybe function's comment and example result will be useful. Because what I expect now based on function's description by calling function
```
$getBlockSpanDetailedList( UUID, 100, 1, 100, true) = {
  'list': [ { startBlock: 99, endBlock: 100}, ... { startBlock: 0, endBlock: 1}],
  'statistics': { // here goes nbdr statistics for range  [0 .. 100]
     mean: someMean,
     stdavg: someAvg
  }
}
```

But instead, I see that I will get:
```
[ (list of 100 elements of type  BlockSpanDetails without statistics) ] + [ (list of 100 elements of type BlockSpanDetail and statistics)]
```
such results confuses me a lot and I don't see explanations in the comments for example of why I don't see result expected by me but I will get result, that consist of 2 lists: one without statistics and another with statistic. So I don't understand how to treat such result at the end